airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcV1V2Migrator.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.integrations.destination.jdbc.typing_deduping;
+
+import io.airbyte.cdk.db.jdbc.JdbcDatabase;
+import io.airbyte.cdk.integrations.destination.NamingConventionTransformer;
+import io.airbyte.cdk.integrations.destination.jdbc.TableDefinition;
+import io.airbyte.integrations.base.destination.typing_deduping.BaseDestinationV1V2Migrator;
+import io.airbyte.integrations.base.destination.typing_deduping.NamespacedTableName;
+import io.airbyte.integrations.base.destination.typing_deduping.StreamConfig;
+import java.sql.ResultSet;
+import java.util.Collection;
+import java.util.Optional;
+import lombok.SneakyThrows;
+
+/**
+ * Largely based on
+ * {@link io.airbyte.integrations.destination.snowflake.typing_deduping.SnowflakeV1V2Migrator}.
+ */
+public class JdbcV1V2Migrator extends BaseDestinationV1V2Migrator<TableDefinition> {
+
+  private final NamingConventionTransformer namingConventionTransformer;
+  private final JdbcDatabase database;
+  private final String databaseName;
+
+  public JdbcV1V2Migrator(final NamingConventionTransformer namingConventionTransformer, final JdbcDatabase database, final String databaseName) {
+    this.namingConventionTransformer = namingConventionTransformer;
+    this.database = database;
+    this.databaseName = databaseName;
+  }
+
+  @SneakyThrows
+  @Override
+  protected boolean doesAirbyteInternalNamespaceExist(final StreamConfig streamConfig) throws Exception {
+    String retrievedSchema = "";
+    try (ResultSet columns = database.getMetaData().getSchemas(databaseName, streamConfig.id().rawNamespace())) {
+      while (columns.next()) {
+        retrievedSchema = columns.getString("TABLE_SCHEM");
```

Comment:
is this a typo? `TABLE_SCHEM` vs `TABLE_SCHEMA`

Reply:
Nope, used it from javadoc 
```
Retrieves the schema names available in this database. The results are ordered by TABLE_CATALOG and TABLE_SCHEM.
The schema columns are:
TABLE_SCHEM String => schema name
TABLE_CATALOG String => catalog name (may be null)
```